core/storage/story/gae_models.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models for storing the story data models."""
+
+from core.platform import models
+
+from google.appengine.ext import ndb  # pylint: disable=relative-import
+
+(base_models, user_models,) = models.Registry.import_models([
+    models.NAMES.base_model, models.NAMES.user])
+
+
+class StorySnapshotMetadataModel(base_models.BaseSnapshotMetadataModel):
+    """Storage model for the metadata for a story snapshot."""
+    pass
+
+
+class StorySnapshotContentModel(base_models.BaseSnapshotContentModel):
+    """Storage model for the content of a story snapshot."""
+    pass
+
+
+class StoryModel(base_models.VersionedModel):
+    """Model for storing stories.
+
+    This class should only be imported by the story domain file, the
+    story services file, and the story model test file.
+    """
+    SNAPSHOT_METADATA_CLASS = StorySnapshotMetadataModel
+    SNAPSHOT_CONTENT_CLASS = StorySnapshotContentModel
+    ALLOW_REVERT = False
+
+    # The title of the story.
+    title = ndb.StringProperty(required=True, indexed=True)
+    # A high-level description of the story.
+    description = ndb.StringProperty(indexed=False)
+    # A set of notes, that describe the characters, main storyline, and setting.
+    notes = ndb.TextProperty(indexed=False)
+    # The ISO 639-1 code for the language this question is written in.
+    language_code = ndb.StringProperty(required=True, indexed=True)
+    # The schema version for the story_contents.
+    schema_version = (
+        ndb.IntegerProperty(required=True, default=1, indexed=True))
+    # The story graph specifying the connection between nodes.
```

Comment:
Give the reader a pointer to where the definition of the story_contents are.

That said, I tried to hunt for this and couldn't find it in the domain layer. Shouldn't we have a domain object for StoryContents?

Reply:
Actually, that is the same as StoryNode. story_contents is just a list of StoryNode objects.

Reply:
According to this it's a dict with one key (nodes). https://github.com/oppia/oppia/pull/4953/files#diff-f3c38ab639b34a337bb09e9612d8efe3R223

So it should be a domain object with one field, right? I know it seems trivial for now but the point is that we may later add other stuff to that dict.

Reply:
So, you mean, to have a dict in story_domain called story_contents instead of story_nodes and have it just like that in the storage level?

Reply:
Not quite. Have a domain object in story_domain called StoryContents, the same way that you would represent any sub-dict of the top-level object.

Reply:
Yeah, so StoryContents would have a single element nodes, which would be a list of StoryNode. In that case, can't we replace the story_nodes in the Story domain object with this story_contents?